test/integration/smart_proxy_test.rb

```diff
@@ -39,4 +39,13 @@ class SmartProxyIntegrationTest < ActionDispatch::IntegrationTest
     assert page.has_selector?('h3', :text => "DHCP")
     assert page.has_content? "Version"
   end
+
+  test "show page with conditional pagelets" do
+    Pagelets::Manager.add_pagelet("smart_proxies/show", :main_tabs, :name => "Visible", :partial => "../../test/static_fixtures/views/test", :onlyif => Proc.new { true })
+    Pagelets::Manager.add_pagelet("smart_proxies/show", :main_tabs, :name => "Hidden", :partial => "../../test/static_fixtures/views/test", :onlyif => Proc.new { false })
+    proxy = smart_proxies(:one)
+    visit smart_proxy_path(proxy)
+    assert page.has_link? "Visible"
+    refute page.has_link? "Hidden"
+  end
```

Comment:
Could you please add more tests for both simple and tab pagelets? It's also worth testing whether the model is actually passed to the `:onlyif` proc.